### ISSUE TITLE
[documentation] Document that different function apps sharing same storage account need different task hubs

### ISSUE THREAD

**Issue by anthonychu** [Azure/azure-functions-durable-extension#1066]
Ran into this in a support case triage today and yesterday while chatting with @simonaco.

Some customers create a single storage account and share it with multiple apps locally and/or in the cloud. Sometimes this works, but this doesn't work with Durable Functions and task hubs.

Need to document better that storage accounts shouldn't be shared, unless using different task hubs(?).

**Comment by ConnorMcMahon** [comment]
I know that we have this documented in at least one place already. For instance on the second paragraph of the [task hub concept page](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-task-hubs).

> If multiple function apps share a storage account, each function app must be configured with a separate task hub name. A storage account can contain multiple task hubs. The following diagram illustrates one task hub per function app in shared and dedicated storage accounts.

That being said, this is the 13th page under _Concepts_, and customers probably shouldn't need to read that far for such a simple gotchya. I'm not sure the best way to highlight this further, but am open to ideas.

Maybe having a simple one page document we can link people to for the most important things to know before starting to develop for Durable (i.e. no non-determinism in orchestrations, no shared task hubs, etc). One of the complaints we often get is that customers feel they need to be an expert to use Durable. If we condensed the most useful advice that covers 80% of scenarios in a single simple document, that could reduce the burden. We could add links to the relevant concept sections to elaborate why these practices must be followed.
